wpiunits/src/main/java/edu/wpi/first/units/Measure.java

```diff
@@ -265,6 +265,30 @@ default boolean isNear(Measure<?> other, double varianceThreshold) {
     return Math.abs(this.baseUnitMagnitude() - other.baseUnitMagnitude()) <= tolerance;
   }
 
+  /**
+   * Checks if this measure is near another measure of the same unit, with a specified tolerance of
+   * the same unit.
+   *
+   * <pre>
+   *     Meters.of(1).isNear(Meters.of(1.2), Millimeters.of(300)) // true
+   *     Degrees.of(90).isNear(Rotations.of(0.5), Degrees.of(45)) // false
+   * </pre>
+   *
+   * @param other the other measure to compare against.
+   * @param tolerance the tolerance allowed in which the two measures are defined as near each
+   *     other.
+   * @return true if this unit is near the other measure, otherwise false.
+   */
+  default boolean isNear(Measure<?> other, Measure<?> tolerance) {
```

Comment:
Let's use `Measure<U>` here instead of wildcards. The wildcards can be useful if you check unit compatibility (rather than only allowing measures of the same dimension), but comparing the  base unit objects is just a worse runtime check.

Reply:
22a1afa6400fe10c475f213741e3dc33dde24116
ef5a722b834b752cfca52e0f555ebf356a88b9eb